internal/controllers/synthesis/lifecycle.go

```diff
@@ -177,11 +179,18 @@ func (c *podLifecycleController) Reconcile(ctx context.Context, req ctrl.Request
 	sytheses.Inc()
 
 	// This metadata is optional - it's safe for the process to crash before reaching this point
-	copy := comp.DeepCopy()
-	copy.Status.CurrentSynthesis.Attempts++
-	copy.Status.CurrentSynthesis.PodCreation = &pod.CreationTimestamp
-	err = c.client.Status().Patch(ctx, copy, client.MergeFrom(comp))
+	patch := []map[string]any{
+		{"op": "test", "path": "/status/currentSynthesis/uuid", "value": comp.Status.CurrentSynthesis.UUID},
```

Comment:
Essentially moving these checks to the apiserver instead of doing them locally?

Reply:
Exactly. The problem is that the normal optimistic concurrency control approach is... too optimistic. i.e. it's very likely that the resource version has changed at this point, but it's also likely that the resource itself hasn't changed in ways that _semantically_ conflict with this patch. So the `test` op allows us to preserve safety in extreme cases without sacrificing efficiency in the happy path